packages/local_auth/local_auth_ios/ios/Classes/FLTLocalAuthPlugin.m

```diff
@@ -147,9 +147,13 @@ - (nullable NSNumber *)deviceCanSupportBiometricsWithError:
 
 - (nullable NSNumber *)isDeviceSupportedWithError:
     (FlutterError *_Nullable __autoreleasing *_Nonnull)error {
-  // TODO(stuartmorgan): Fix this to check for biometrics or passcode; see
-  // https://github.com/flutter/flutter/issues/116179
-  return @YES;
+  LAContext *context = [[LAContext alloc] init];
+  NSError *authError = nil;
+  if ([context canEvaluatePolicy:LAPolicyDeviceOwnerAuthentication error:&authError]) {
+    return @YES;
+  } else {
+    return @NO;
+  }
```

Comment:
You can box expressions in addition to literals; this whole block is equivalent to `return @(the current if condition);`.